Network/Units/Client/Client.cpp

```diff
@@ -6,3 +6,138 @@
 */
 
 #include "Client.hpp"
+
+#include <iostream>
+#include <iomanip>
+
+namespace ABINetwork
+{
+
+Client::Client()
+{
+    _roomPassword = "";
+    _loginState = LoginState::NONE;
+    _serverIp = "";
+    _tcpPort = 0;
+    _udpPort = 0;
+    _io_context = std::make_shared<asio::io_context>();
+    if (!_io_context)
+        throw ABIError("Failed to create the client");
+    _tcp_socket = std::make_shared<asio::ip::tcp::socket>(*_io_context);
+    _udp_socket = std::make_shared<asio::ip::udp::socket>(*_io_context);
+    if (!_tcp_socket || !_udp_socket)
+        throw ABIError("Failed to create the client");
+    _server_endpoint = nullptr;
+    _messageId = 0x0000;
+    _token = 0;
+    _io_thread = nullptr;
+}
+
+Client::~Client()
+{
+    _io_context->stop();
```

Comment:
It doesn't stop by itself ?

Reply:
No there are some problems with ASIO.
Sometimes when you leave the client it freeze because ASIO's threads.